anoncreds/src/data_types/anoncreds/schema.rs

```diff
@@ -1,67 +1,23 @@
-use crate::data_types::identifiers::schema::SchemaId;
-use crate::data_types::utils::Qualifiable;
 use crate::data_types::{Validatable, ValidationError};
+use crate::impl_anoncreds_object_identifier;
 
 use std::collections::HashSet;
 use std::iter::FromIterator;
 
 pub const MAX_ATTRIBUTES_COUNT: usize = 125;
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-#[serde(tag = "ver")]
-pub enum Schema {
-    #[serde(rename = "1.0")]
-    SchemaV1(SchemaV1),
-}
-
-impl Schema {
-    pub fn id(&self) -> &SchemaId {
-        match self {
-            Schema::SchemaV1(s) => &s.id,
-        }
-    }
-
-    pub fn to_unqualified(self) -> Schema {
-        match self {
-            Schema::SchemaV1(schema) => Schema::SchemaV1(SchemaV1 {
-                id: schema.id.to_unqualified(),
-                name: schema.name,
-                version: schema.version,
-                attr_names: schema.attr_names,
-                seq_no: schema.seq_no,
-            }),
-        }
-    }
-}
-
-impl Validatable for Schema {
-    fn validate(&self) -> Result<(), ValidationError> {
-        match self {
-            Schema::SchemaV1(schema) => schema.validate(),
-        }
-    }
-}
+impl_anoncreds_object_identifier!(SchemaId);
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct SchemaV1 {
-    pub id: SchemaId,
+pub struct Schema {
     pub name: String,
     pub version: String,
-    #[serde(rename = "attrNames")]
```

Comment:
Let's wait with this one for a bit maybe. I know I created an issue to change it. But as all models are coming out to use camelCase, I think we should just jeep using camelCase here and minimize changes.

Thoughts?

Reply:
I am fine with either. I think a definitive decision would be very valuable here. 

@swcurran I might have missed some of the on going discussions surrounding the spec, but is there a decision made around the anoncreds objects and which casing they use? I think I have a slight preference for snake case, but without any good reason, so either is good in my opinion. (if this needs further discussion, and there is no issue open already, I would not mind creating one).

Reply:
The reason I'm leaning to keeping it camelCase is that anoncreds objects already mostly use camelCase. The records that are registered on the indy ledger are using snake_case in some places however.

So my opinion would be to go with camelCase